src/server_manager/web_app/ui_components/outline-server-view.js

```diff
@@ -801,43 +803,61 @@ export class ServerView extends DirMixin(PolymerElement) {
     }
   }
 
-  _formatBytesTransferred(numBytes, emptyValue = '') {
+  _makeReadableDataAmount(amount, numDecimals) {
+    const readable = byte_size(amount, {precision: numDecimals});
+    return {
+      amount: readable.value,
+      // Make the unit singular
+      unit: readable.long.slice(0, -1)
+    }
+  }
+
+  _getFormatPrecision(numBytes) {
+    if (numBytes >= 10 ** 9) {
+      return 2;
+    } else if (numBytes >= 10 ** 6) {
+      return 1;
+    }
+    return 0;
+  }
+
+  _formatBytesTransferred(numBytes, language, emptyValue = '') {
     if (!numBytes) {
       // numBytes may not be set for manual servers, or may be 0 for
       // unused access keys.
       return emptyValue;
     }
 
     // Show 0 decimals for < 1MB, 1 decimal for >= 1MB, 2 decimals for >= 1GB.
-    let numDecimals = 0;
-    if (numBytes >= 10 ** 9) {
-      numDecimals = 2;
-    } else if (numBytes >= 10 ** 6) {
-      numDecimals = 1;
-    }
-
-    return byte_size(numBytes, {precision: numDecimals}).toString();
+    const data = this._makeReadableDataAmount(numBytes, this._getFormatPrecision(numBytes));
+    return i18n.makeUnitFormatter(data.unit, language).format(data.amount);
   }
 
   _getFormattedTransferredUnit(numBytes, emptyValue = '') {
-    const formattedTransfer = this._formatBytesTransferred(numBytes);
-    if (!formattedTransfer) {
-      return emptyValue;
-    }
-    const units = formattedTransfer.match(/[a-zA-Z%]+/g);
-    if (!units || units.length < 0) {
+    if (!numBytes) {
       return emptyValue;
     }
-    return units.pop();
+    const data = this._makeReadableDataAmount(numBytes, 0);
+    return i18n.formattedUnit(data.unit, this.language);
   }
 
   _getFormattedTransferredValue(numBytes, emptyValue = '') {
-    const formattedTransfer = this._formatBytesTransferred(numBytes);
-    if (!formattedTransfer) {
+    if (!numBytes) {
       return emptyValue;
     }
-    const value = parseFloat(formattedTransfer);
-    return !!value ? value : emptyValue;
+    const data = this._makeReadableDataAmount(numBytes, this._getFormatPrecision(numBytes));
+    const parts = i18n.makeUnitFormatter(data.unit).formatToParts(data.amount);
+    // Assumes that the last two parts are a space and the unit
+    return parts.slice(0, -2).map(part => part.value).join('');
```

Comment:
You can't assume that. Look for `type === 'unit'` instead.

Reply:
Done in 79fb1ea